### PR TITLE
Added callsites for GauntletPartyScreen

### PR DESCRIPTION
**Pull request by tbeswick96**
- Added `OnActivate` patch callsite for GauntletPartyScreen
- Allows creation of PartyVM mixins

### CODE REVIEW

**Comment by MLNW** [comment]
That's exactly the screen that I've been trying to extend for a few days now. I stumped my head on how to resolve the interface methods via Harmony. This helps tremendously. I'd love to see this merged, if functional.

**Comment by tbeswick96** [comment]
@MLNW what mod are you making? I've added this for some party menu sorting and management functionality

**Comment by MLNW** [comment]
> @MLNW what mod are you making? I've added this for some party menu sorting and management functionality

I wanted to add a simple button to sort the troops as a first taste of how to mod Bannerlord. I've always wanted that functionality in Warband but since you could only load one mod at a time it wasn't worth looking into it.

I've already implemented sorting by level but by misusing the Reset button atm and without actually refreshing the UI since I haven't found access to that particular functionality.

File: UIExtenderLibModule/UIExtenderLibModule.csproj [hunk 1]
```diff
@@ -172,6 +172,8 @@
     <Compile Include="UIExtenderLibModule.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="Utils.cs" />
+    <Compile Include="ViewModel\Patches\GauntletPartyScreenCallsites.cs" />
+    <Compile Include="ViewModel\Patches\CharacterDeveloperVMCallsites.cs" />
```
Comment (MLNW) [inline UIExtenderLibModule/UIExtenderLibModule.csproj:176]: For some reason you reference another callsites class that is not part of this commit.

```suggestion

File: UIExtenderLib.sln.DotSettings [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	<s:String x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/OTHER_BRACES/@EntryValue">NEXT_LINE</s:String>
+	<s:String x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/TYPE_DECLARATION_BRACES/@EntryValue">NEXT_LINE</s:String>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpKeepExistingMigration/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpPlaceEmbeddedOnSameLineMigration/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpUseContinuousIndentInsideBracesMigration/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ESettingsUpgrade_002EMigrateBlankLinesAroundFieldToBlankLinesAroundProperty/@EntryIndexedValue">True</s:Boolean></wpf:ResourceDictionary>
```
Comment (MLNW) [inline UIExtenderLib.sln.DotSettings:13]: Sounds like this is a user config file that should not be part of the repository. At least not added during an unrelated pull request.

File: UIExtenderLibModule/ViewModel/Patches/GauntletPartyScreenCallsites.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+namespace UIExtenderLibModule.ViewModel.Patches
+{
+    //  Harmony cannot resolve the interface methods by iself, so this route has to be used
+    [HarmonyPatch]
+    public static class GauntletPartyScreenOnActivateCallsite
+    {
+        public static MethodBase TargetMethod()
+        {
+            Type type = typeof(GauntletPartyScreen);
+            MethodInfo interfaceMethod = type.GetInterface(nameof(IGameStateListener)).GetMethod("OnActivate");
```
Comment (MLNW) [inline UIExtenderLibModule/ViewModel/Patches/GauntletPartyScreenCallsites.cs:18]: There is also an `OnInitialize` method that may be a better point to use as it is most likely called once instead of multiple times. As far as I understand the library as of now this only needs to be done once.

**Review by MLNW — CHANGES_REQUESTED** [review]
Seems like there are some unrelated changes and also some room for improvement.

File: UIExtenderLib.sln.DotSettings (hunk 2 above)
Comment (tbeswick96) [reply UIExtenderLib.sln.DotSettings:13]: It's not a user config file.
It's code style rules for rider to ensure styling consistency across contributers. No harm in adding them and doesn't warrant its own PR

File: UIExtenderLibModule/UIExtenderLibModule.csproj (hunk 1 above)
Comment (tbeswick96) [reply UIExtenderLibModule/UIExtenderLibModule.csproj:176]: See https://github.com/shdwp/UIExtenderLib/pull/1

The file is not added to the project file correctly in this PR. Consequence of including it correctly in my local project version. 
Feel free to PR the change separately.

File: UIExtenderLibModule/ViewModel/Patches/GauntletPartyScreenCallsites.cs (hunk 3 above)
Comment (tbeswick96) [reply UIExtenderLibModule/ViewModel/Patches/GauntletPartyScreenCallsites.cs:18]: No, OnInitialize does not fire correctly for the first time the party menu is opened. I have already explored this. It does fire each subsequent time the menu is opened. I can add the hook here for it back in, but for my purposes the various sorting functions occur on button presses, rather than on menu open

**Comment by tbeswick96** [comment]
Yes so far in my implementation I have sorting by name, type, group, and tier. I also have an option to put upgradable troops at the top of the list.

UI update causes game freeze, and I've found no way around this so far, hence usage of buttons to trigger sorting, rather than having it on menu open. 

I'm currently trying to figure out how to get the last used sort mode saving with the built in saving rather than as a config file, and once I've done that I'll get my stuff up on github so you can contribute there if you like. 

About 4 other sorting mods have popped up in the last day, and I imagine later down the line they will merge into one feature complete mod

**Comment by MLNW** [comment]
> I'm currently trying to figure out how to get the last used sort mode saving with the built in saving rather than as a config file, and once I've done that I'll get my stuff up on github so you can contribute there if you like.

Let me know.

> About 4 other sorting mods have popped up in the last day, and I imagine later down the line they will merge into one feature complete mod

Sounds about right. If integrated with the UI this would be a huge UX improvement.

**Comment by tbeswick96** [comment]
@MLNW Do you have discord? Feel free to PM me on there if you're interested in helping out with dev TimB#0156

**Comment by shdwp** [comment]
Thanks for the contribution! Good suggestion about handling patches more carefully, after callsite patch codebase grows this will surely be a problem to keep them all updated. I will probably rewrite the patches interface after this PR.
